corehq/apps/reports/filters/controllers.py

```diff
@@ -122,10 +123,19 @@ def _get_users(self, query, start, size, include_inactive=False):
         users = customize_user_query(self.request.couch_user, self.domain, users)
         return [self.utils.user_tuple(u) for u in users.run().hits]
 
+    def _should_include_enterprise_controlled_domains(self):
+        from corehq.apps.reports.standard.users.reports import UserHistoryReport
+        if UserHistoryReport.slug in self.request.META['HTTP_REFERER']:
```

Comment:
Nitpick: you could do f"/{UserHistoryReport.slug}/", even though the slug is unlikely to coincidentally appear elsewhere in the URL.